test/deployer/AccessControlledDeployer.t.sol

```diff
@@ -0,0 +1,372 @@
+    "@axelar-network/axelar-gmp-sdk-solidity/contracts/test/token/ERC20MintableBurnableInit.sol";
+
+import {OwnableCreate2Deployer} from "../../contracts/deployer/create2/OwnableCreate2Deployer.sol";
+import {AccessControlledDeployer} from "../../contracts/deployer/AccessControlledDeployer.sol";
+import {OwnableCreate3Deployer} from "../../contracts/deployer/create3/OwnableCreate3Deployer.sol";
+
+import {Create2Utils} from "./create2/Create2Utils.sol";
+import {Create3Utils} from "./create3/Create3Utils.sol";
+
+contract AccessControlledDeployerTest is Test, Create2Utils, Create3Utils {
+    address private admin = makeAddr("admin");
+    address private pauser = makeAddr("pauser");
+    address private unpauser = makeAddr("unpauser");
+    address[] private authDeployers;
+    AccessControlledDeployer private rbacDeployer;
+
+    event Deployed(address indexed deployedAddress, address indexed sender, bytes32 indexed salt, bytes32 bytecodeHash);
+
+    error ZeroAddress();
+    error EmptyDeployerList();
+    error NotOwnerOfDeployer();
+
+    function setUp() public {
+        rbacDeployer = new AccessControlledDeployer(admin, pauser, unpauser);
+
+        authDeployers.push(makeAddr("deployer1"));
+        vm.prank(admin);
+        rbacDeployer.grantDeployerRole(authDeployers);
+    }
+
+    /**
+     * Constructor
+     */
+    function test_Constructor_RevertIf_AdminIsZeroAddress() public {
+        vm.expectRevert(ZeroAddress.selector);
+        new AccessControlledDeployer(address(0), pauser, unpauser);
+    }
+
+    function test_Constructor_RevertIf_PauserIsZeroAddress() public {
+        vm.expectRevert(ZeroAddress.selector);
+        new AccessControlledDeployer(admin, address(0), unpauser);
+    }
+
+    function test_Constructor_RevertIf_UnpauserIsZeroAddress() public {
+        vm.expectRevert(ZeroAddress.selector);
+        new AccessControlledDeployer(admin, pauser, address(0));
+    }
+
+    function test_Constructor_AssignsRoles() public {
+        assertTrue(rbacDeployer.hasRole(rbacDeployer.DEFAULT_ADMIN_ROLE(), admin));
+        assertTrue(rbacDeployer.hasRole(rbacDeployer.PAUSER_ROLE(), pauser));
+        assertTrue(rbacDeployer.hasRole(rbacDeployer.UNPAUSER_ROLE(), unpauser));
+    }
+
+    /**
+     * Admin role management
+     */
+    function test_AdminCanAssignRoles() public {
+        address newPauser = makeAddr("newPauser");
+        address newUnpauser = makeAddr("newUnpauser");
+
+        vm.startPrank(admin);
+        rbacDeployer.grantRole(rbacDeployer.PAUSER_ROLE(), newPauser);
+        rbacDeployer.grantRole(rbacDeployer.UNPAUSER_ROLE(), newUnpauser);
+
+        assertTrue(rbacDeployer.hasRole(rbacDeployer.PAUSER_ROLE(), newPauser));
+        assertTrue(rbacDeployer.hasRole(rbacDeployer.UNPAUSER_ROLE(), newUnpauser));
+    }
+
+    function test_AdminCanRevokeRoles() public {
+        vm.startPrank(admin);
+        rbacDeployer.revokeRole(rbacDeployer.PAUSER_ROLE(), pauser);
```

Comment:
would be nice to assert they have roles before revoking?

Reply:
good point. added.